src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -1320,11 +1321,13 @@ void G1ConcurrentMark::cleanup() {
   double start = os::elapsedTime();
 
   verify_during_pause(G1HeapVerifier::G1VerifyCleanup, VerifyOption_G1UsePrevMarking, "Cleanup before");
-
-  {
+  bool has_rebuilt_remembered_sets = (total_selected_for_rebuild() > 0);
+  if (has_rebuilt_remembered_sets) {
     GCTraceTime(Debug, gc, phases) debug("Update Remembered Set Tracking After Rebuild", _gc_timer_cm);
     G1UpdateRemSetTrackingAfterRebuild cl(_g1h);
     _g1h->heap_region_iterate(&cl);
+  } else {
+    log_debug(gc, remset, tracking)("No Remembered Sets To Update After Rebuild");
```

Comment:
Maybe make this `gc+phases=debug` too like the one above so that either of those shows up with the same log options. Also there does not seem to be a reason for the upper casing here. (The argument to `GCTraceTime` is a phase name, but this seems like a regular statement)